src/Array/extensions.js

```diff
@@ -50,6 +50,30 @@ extensionArray.remove = function remove(selector) {
     return this;
 };
 
+/**
+ * Remove all instances of a value within the array and return the removed value
+ * @param {*} value - The value trying to match against
+ */
+extensionArray.yank = function yank(value){
+    if (value.constructor === Array) {        
+        for (let i = 0; i < value.length; i++) {
+            if(this.includes(value[i])){ 
+               this.remove(value[i]);
+            }else{
+                // Remove it from the given array so when we return it, it doesnt 
+                // return a number that was not removed
+                value.remove(value[i]);
+            }                
+        }
+        return value;
+    } else if (this.includes(value)) {
+        this.remove(value);
+        return value;
+    } else{
+        throw "Value not in the array";
```

Comment:
I think it'd be better to just return a `null` if there is nothing found in the yank. That way if it was going to be used in a `if` statement, or a tertiary statement it could be used like `if(foo.yank(2))`. Here it'd just throw a crazy uncaught error.